src/operator/slice_channel-inl.h

```diff
@@ -172,15 +173,35 @@ class SliceChannelProp : public OperatorProperty {
       << ") does not divide input dimension "
       << real_axis << " (" << dshape[real_axis] << ").";
     dshape[real_axis] /= param_.num_outputs;
-    if (param_.squeeze_axis && dshape[real_axis] == 1) {
+    if (param_.squeeze_axis && (dshape[real_axis] == 1 || ishape[real_axis] == 0)) {
```

Comment:
OK, I find that the current logic is that if (squeeze_axis=true && shape becomes 1), squeeze the dimension... However it's unnecessary to use squeeze_axis if the ndim is unchanged.